staging/src/k8s.io/apiserver/pkg/cel/environment/base.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package environment
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/ext"
+
+	celconfig "k8s.io/apiserver/pkg/apis/cel"
+	"k8s.io/apiserver/pkg/cel/library"
+)
+
+// WARNING: Do not change to publishedEnvOpts or publishedProgramOpts
+// without first applying the change to storageEnvOpts or storageProgramOpts
+// for at least one Kubernetes minor release.
+// All changes must be backward compatible.
+
+var publishedEnvOpts = []cel.EnvOption{
+	cel.HomogeneousAggregateLiterals(),
+	// Validate function declarations once during base env initialization,
+	// so they don't need to be evaluated each time a CEL rule is compiled.
+	// This is a relatively expensive operation.
+	cel.EagerlyValidateDeclarations(true),
+	cel.DefaultUTCTimeZone(true),
+}
+
+var publishedProgramOpts = []cel.ProgramOption{
+	cel.EvalOptions(cel.OptOptimize, cel.OptTrackCost),
+	cel.CostLimit(celconfig.PerCallLimit),
+}
+
+var publishedLibs = []cel.EnvOption{
+	ext.Strings(),
+	library.URLs(),
+	library.Regex(),
+	library.Lists(),
+	library.Authz(),
+}
+
+// Modify these variables when adding language configuration or libraries that are planned to
+// made available in a future Kubernetes release.
+var storageEnvOpts = publishedEnvOpts
+var storageProgramOpts = publishedProgramOpts
+
+var storageLibs = append(publishedLibs[0:len(publishedLibs):len(publishedLibs)],
+	library.Test(), // TODO: remove
```

Comment:
This is referring to removing this particular `append` and not the entire `Test` library, right? I figure it's the former case but just wanted to double-check.

Reply:
I may keep this function here long term. I'm still trying to decide if it is better to keep this lib in storage or to do a bunch more DI wiring. See the PR description.  For now I'll replace this TODO with a comment.